h2o-core/src/main/java/water/ExternalFrameWriterClient.java

```diff
@@ -214,15 +215,10 @@ private void waitForRequestToFinish(int timeout, byte confirmation) throws Exter
     }
   }
 
-  private void increaseCurrentColIdx() throws IOException, ExternalFrameConfirmationException {
+  private void increaseCurrentColIdx() {
     currentColIdx++;
     if (currentColIdx == numCols) {
       currentRowIdx++;
-      if (currentRowIdx == numRows) {
-        writeToChannel(ab, channel);
```

Comment:
there is nothing to write to channel as we write data directly as part of `send` methods. This invocation can be safely removed